test/test_restful.py

```diff
@@ -276,6 +277,33 @@ def initthread():
     return True
 
 
+def launch_rust_cloudagent():
+    """Start up the cloud agent"""
+    global agent_process, script_env, FORK_ARGS
+    if agent_process is None:
+        agent_process = subprocess.Popen("/.cargo/bin/keylime_agent",
+                                         shell=False,
+                                         preexec_fn=os.setsid,
+                                         stdout=subprocess.PIPE,
+                                         stderr=subprocess.STDOUT,
+                                         env=script_env)
+
+        def initthread():
+            sys.stdout.write('\033[94m' + "\nRust Cloud Agent Thread" + '\033[0m')
+            while True:
+                line = agent_process.stdout.readline()
+                if line == b'':
+                    break
+                # line = line.rstrip(os.linesep)
+                line = line.decode('utf-8')
+                sys.stdout.flush()
+                sys.stdout.write('\n\033[94m' + line + '\033[0m')
+        t = threading.Thread(target=initthread)
+        t.start()
+        time.sleep(10)
```

Comment:
I guess that this sleep is here to wait for the agent to do the bind. This can produce flaky tests (tests that sometimes fails).

Maybe a better option is to wait (not tested):

```python
for retry in range(10):
    r = subprocess.run(["ss", "-l", "-t", "-n", "-p",  "( sport = :9002 )"], stdout=subprocess.PIPE)
    if b':9002" in r.stdout:
        break
    time.sleep(1)
else:
   raise Exception("Rust keylime_agent failed to run")
```

Reply:
This helper was copy-pasted from the `launch_cloudagent` function, which also does it this way. Likewise for the 10-second wait after initialization.

I'm not opposed to changing it, but it would also make sense to do it for the Python launch code too.